### PR TITLE
Render Local Time

### LINKED ISSUE #5134: Local Time Display Component
## What type(s) of user does this feature affect?
 - volunteers
 - supervisors
 - admins
 - all casa admins

## Description
Displayed time across the app has timezones determined by the server but the user may not have the same time zone as the server resulting in incorrect times. In some cases the time displayed is actually UTC time.

Create a [view component](https://viewcomponent.org/guide/getting-started.html) that renders the local time ~html with a unix timestamp and converts it to the local time on the client via javascript. Similar to [discord timestamps](https://www.pattlesstore.com/articles/how-to-use-timestamps-on-discord)~

The component should take as arguments
 - the time in unix timestamp format
 - the format

The [`title`](https://www.w3schools.com/TAGS/att_title.asp) of the component should show a more specific representation of the time.

~so on the page it will render something like <span><t: UNIX_TIME_VALUE, f: FORMAT ></span> and then javascript will convert the html into something human readable~ it's probably better to get the time zone from the cookies from the browser so no js required

Write Tests

See https://github.com/rubyforgood/casa/wiki/Working-with-Frontend-Assets

## Screenshots of current behavior, if any
You can paste images on the clipboard here

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
### What github issue is this PR for, if any?
Resolves #5134

### What changed, and why?
We created a new view component named LocalTimeComponent. This component renders user local time.

### How is this tested? (please write tests!) 💖💪
Component is tested with both positive and negative tests.

### CODE REVIEW

File: app/components/local_time_component.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    formatted_date + " at " + time_of_day + " " + time_zone
+  end
+
+  private
+
+  def before_render
+    browser_tz = ActiveSupport::TimeZone.find_tzinfo(cookies[:browser_time_zone])
+    ActiveSupport::TimeZone.all.find { |zone| zone.tzinfo == browser_tz } || Time.zone
+  rescue TZInfo::UnknownTimezone, TZInfo::InvalidTimezoneIdentifier
+    @time_zone = Time.zone
```
Comment (FireLemons) [inline app/components/local_time_component.rb:32]: we need this from the client. Can you add this as a parameter for the component?

**Comment by FireLemons** [comment]
Can you add this to a random view in your local app without committing and take a screenshot?

**Review by FireLemons — CHANGES_REQUESTED** [review]

File: app/components/local_time_component.rb (hunk 1 above)
Comment (SajjadAhmad14) [reply app/components/local_time_component.rb:32]: Hi @FireLemons, this is the local time zone of the user taken from browser. The arguments required for the component and mentioned in the issue are the time in unix timestamp format, and the format. So do I need to pass the time zone as the third argument?

**Comment by SajjadAhmad14** [comment]
> Can you add this to a random view in your local app without committing and take a screenshot?
time

  <%= render (LocalTimeComponent.new(format: 12, unix_timestamp: 1694349775))%>
I rendered this component in users/edit_profile.rb

File: app/components/local_time_component.rb (hunk 1 above)
Comment (FireLemons) [reply app/components/local_time_component.rb:32]: Yes, it's from the client

**Comment by FireLemons** [comment]
Oh my bad. I didn't know `Time.zone` was from the client. :1st_place_medal:

**Pull request merged**